### DISCUSSION TITLE
Generator template

### DISCUSSION THREAD

**Discussion by oottoohh** [infinitered/ignite#2409] [Q&A] [answered]
i see new feature on v8.6.0, generrator can generate template to sub directory. i need this on my old version ignite. what i must going todo ? i try following changes on v8.6.0 but it doesnt work ..thanks advance !

**Comment by frankcalise** [comment]
Hey @oottoohh!

You can grab the latest Ignite templates from the repository and simply just update the content in your `ignite/templates` directory within your project.

You can see an example of the subdirectory change to the template in the component one here: https://github.com/infinitered/ignite/blob/master/boilerplate/ignite/templates/component/NAME.tsx.ejs

**Comment by oottoohh** [comment]
Thanks for response @frankcalise 
I already try this yesterday but still not working sir, on my project i have custom ignite template from default boilerplate and custom absolute path before v6.8.0 release, so from my custom template i try adjust to new enhancement ignite v6.8.0 by just adding :

`append: "export * from \"./<%= props.subdirectory %><%= props.pascalCaseName %>\"\n"`

**Reply by frankcalise** [reply]
Hey would you mind pasting your template (or some variation of it if it can't be shared) and also the command you are running please?

**Reply by oottoohh** [reply]
there is no different code file with default template, the different is i have custom absolute path import using path props from compiler option inside tsconfig then i just split ignite/template file. so i have 2 component file i call this composite and material file. component file from default boilerplate to be materials file.

```destinationDir: app/components/materials
patch:
  path: "app/components/materials/index.ts"
  append: "export * from \"./<%= props.subdirectory %><%= props.pascalCaseName %>\"\n"
  skip: <%= props.skipIndexFile %>
  
import * as React from "react"
import { StyleProp, TextStyle, View, ViewStyle } from "react-native"
import { observer } from "mobx-react-lite"
import { colors, typography } from "@app/theme"
import { Text } from "./Text"

export interface <%= props.pascalCaseName %>Props {
  /**
   * An optional style override useful for padding & margin.
   */
  style?: StyleProp<ViewStyle>
}

/**
 * Describe your component here
 */
export const <%= props.pascalCaseName %> = observer(function <%= props.pascalCaseName %>(props: <%= props.pascalCaseName %>Props) {
  const { style } = props
  const $styles = [$container, style]

  return (
    <View style={$styles}>
      <Text style={$text}>Hello Component Material</Text>
    </View>
  )
})

const $container: ViewStyle = {
  justifyContent: "center",
}

const $text: TextStyle = {
  fontFamily: typography.primary.normal,
  fontSize: 14,
  color: colors.palette.primary500,
}
```

**Reply by oottoohh** [reply]
my command to generate with sub directory same like https://github.com/infinitered/ignite/pull/2285

**Reply by frankcalise** [reply]
So you're trying to generate files into `app/components/materials` and `app/components/composite`? It would be helpful to know what the end goal is.

**Reply by oottoohh** [reply]
@frankcalise correct, actually for now i can generate just file into both folder, but when i try with subdirectory is not working

**Comment by oottoohh** [comment, marked as answer]
i found solution, i just need update my ignite cli to v8.6.1

**Reply by frankcalise** [reply]
Great glad it worked. This is why I previously asked for the command you were running. I'd uninstall it from your global packages and execute it via `npx ignite-cli` in the future to not run into having a stale version.

You can also run the `doctor` or `help` commands to check your version
